pulpcore/cli/core/context.py

```diff
@@ -38,20 +38,29 @@ class PulpArtifactContext(PulpEntityContext):
     HREF = "artifact_href"
     LIST_ID = "artifacts_list"
     READ_ID = "artifacts_read"
+    CREATE_ID = "artifacts_create"
 
-    def upload(self, file: IO[bytes], chunk_size: int = 1000000, check_exists: bool = True) -> Any:
+    def upload(self, file: IO[bytes], chunk_size: int = 1000000) -> Any:
         upload_ctx = PulpUploadContext(self.pulp_ctx)
         start = 0
         size = os.path.getsize(file.name)
-        sha256 = hashlib.sha256()
 
-        if check_exists:
-            for chunk in iter(lambda: file.read(chunk_size), b""):
-                sha256.update(chunk)
-            result = self.list(limit=1, offset=0, parameters={"sha256": sha256.hexdigest()})
-            if len(result) > 0:
-                click.echo("Artifact already exists.", err=True)
-                return result[0]["pulp_href"]
+        sha256 = hashlib.sha256()
+        for chunk in iter(lambda: file.read(chunk_size), b""):
+            sha256.update(chunk)
+        sha256_digest = sha256.hexdigest()
+        file.seek(0)
+
+        if chunk_size > size:
+            # if chunk_size is bigger than the file size, just upload it directly
+            result = self.create({"sha256": sha256_digest}, uploads={"file": file.read()})
+            return result["pulp_href"]
+
+        # check whether the file exists before chunk uploading
+        result = self.list(limit=1, offset=0, parameters={"sha256": sha256_digest})
+        if len(result) > 0:
+            click.echo("Artifact already exists.", err=True)
+            return result[0]["pulp_href"]
```

Comment:
Why not move this check before the atomic upload?

Reply:
I felt like this check was less useful and less worthwhile it for atomic uploads but I suppose it could save users some time if they are uploading big files. Will update.